### DISCUSSION TITLE
Accesso funziona in https da remoto, ma da locale solo in http

### DISCUSSION THREAD

**Discussion by Pippisbruffi** [indomus/forum#1893] [Networking]
Ciao a tutti, ho installato HA OS su Raspberry PI4 da pochi giorni, premetto che come hardware sono ferrato ma tutto ciò che riguarda informatica e codice mi è (per ora) completamente estraneo...
Dunque, seguendo passo passo la guida di indomus sono riuscito a configurare l'accesso da remoto a HA tramite duckdns. Funziona tutto benissimo, compresa l'applicazione android. 
Il problema sorge quando mi connetto utilizzando una rete locale, come ad esempio usando il pc desktop oppure lo smartphone connesso al wifi di casa.
Da browser riesco ad accedere tramite https://homeassistant:8123, ma in alto, accanto all'indirizzo, compare "Non sicuro" con la scritta https barrata in rosso. Non sarebbe nemmeno un problema, se non fosse che in questo modo l'app android di HA non funziona quando sono connesso in locale (anche impostando nelle impostazioni dell'app di utilizzare sempre il dominio di duckdns indipendentemente dalla rete), perchè richiede https.
Cercando online ho visto che avrei potuto risolvere con il proxy NGINX, ho seguito passo passo tutta la procedura di installazione ma non è cambiato niente....
Qualcuno può aiutarmi a fare un po' di troubleshooting per capire dov'è il problema?

Non credo c'entri qualcosa ma specifico che il raspberry può connettersi solo il wifi perchè non ho accesso fisico al modem... Ma posso accedere comunque alla sua pagina di configurazione...

**Comment by snakuzzo** [comment]
Allora...se hai configurato HassOS senza reverse proxy (es.NGINX) in https, HA sia in locale che da remoto risponde SOLO in https. Ed è normale che da remoto sia ok e che in locale dia il warning. Perché c'è bisogno dell'handshake ed il certificato installato vale per il tuo FQDN, non vale per "https://homeassistant.local:8123" (o qualsiasi altro nome o IP utilizzi).

Utilizzando invece il reverse proxy NGINX è normalissimo e GIUSTISSIMO che da remoto ti risponda in https ed in locale in http e ti consiglio di lasciarlo così. l'app companion la configurareai con URL esterna -> https://tuodominio.duckdns.org:tuaporta e URL interna http://tuoip:8123

**Reply by Pippisbruffi** [reply]
Ciao e grazie per la risposta, l'app companion l'ho configurata proprio in quel modo, ma continua a non aprirsi quando sono connesso in rete locale. Sto cercando di capire il problema. Forse qualche impostazione mancante sul router? Per ora ho solo il port mapping da mionome.duckdns.org all'IP del raspberry. 

Inoltre sulla guida dell'addon NGINX dice di verificare che i certificati siano presenti nella cartella /ssl, ma esplorando con Visual Studio Code tale cartella non esiste... Eppure l'accesso da remoto funziona quindi immagino che i certificati siano da qualche parte...

**Reply by snakuzzo** [reply]
Su rete interna non c'entra niente il port mapping / forwarding / o quel che è
Avrai fatto casino...rifai tutto con attenzione

**Reply by Pippisbruffi** [reply]
Dopo diverse prove e imprecazioni, ho deciso di reinstallare HA da zero e provare un altro approccio, ossia con reverse proxy Caddy, seguendo sempre la guida di InDomus. Dopo ulteriori imprecazioni, ho finalmente capito che non riesco a connettermi ad HA se il dispositivo da cui mi connetto è collegato al wifi 2.4Ghz. Se invece è collegato alla 5Ghz riesco a connettermi senza problemi.
Il modem, come la maggior parte dei modem attuali, ha due reti (una da 2.4Ghz e una da 5Ghz) con lo stesso SSID e il band steering attivato.  Al momento ho impostato dal modem che il mio smartphone deve connettersi solo alla 5Ghz e quindi non sto avendo problemi, però mi piacerebbe capire il motivo di questo strano comportamento...
